setup some environment vars for the cron job

Kr1ss-XD opened the pull request:
Some tickets suggest that the mailsync process doesn't have all environ-
ment variables set which are needed. For example:

https://github.com/LukeSmithxyz/mutt-wizard/issues/678
The error messages indicate that for mailsync, `XDG_DATA_HOME` is empty/
unset and hence it cannot resolve the user's gnupg folder.

This commit defines  those variables via the crontab,  which I think are
the relevant ones. Not sure though if these are all we need or if I for-
got some.

Please note, the cronjob itself is still untested because I have no cron
daemon installed on my box. The togglecron function _does_ work properly
though.

LukeSmithxyz commented:
Wait, are you just adding shell variable definitions to the crontab file? Does that actually work?

If it did work, there would also be the problem of if someone toggles the crontab multiple times, there would have a million redundant variable definitions in there since the toggle command doesn't seem to delete them unless I missed something.

Why not add these definitions to the `mailsync` script since it is finding all those other variables anyway?

Kr1ss-XD commented:
> adding shell variable definitions to the crontab file? Does that actually work?

Yes absolutely, with the restriction that cron doesn't resolve variables - it interprets the text to the right of the equal sign literally (after removing leading spaces). E.g. something like `VAR2="$VAR1+sometext"` wouldn't work as expected.
This is no issue here, since variables like `$MBSYNCRC` are expanded in the context of the `mw` script, so they don't end up in the crontab. This behavior is documented in the `crontab(5)` manual.

> The toggle command doesn't seem to delete them unless I missed something.

You didn't miss anything - I did. I had added a leading comment which is used in the `sed` expression, but somehow I've messed up the commit and deleted that comment. That's now fixed.

> Why not add these definitions to the mailsync script since it is finding all those other variables anyway?

I've considered that too, and sure it would be possible.
The reason I used the crontab is I could imagine that users could have those variables defined in a thousand different ways, and they can potentially depend on variables we haven't considered yet. So you might end up adding more and more variables to be detected by `mailsync` when reports come in.

If you'd rather add them to the `eval` command in `mailsync` anyways (or if I'm missing something), I'd be d'accord to close this PR and submit a new one.

LukeSmithxyz commented:
Thinking this through, these default variable definitions should either be without the `${}` or they should be added *after* the `eval` loop in the `mailsync` script. The `${}` is unecessary as it is because we know that those variables haven't been defined already, but I think it would be proper to use them after the `eval` loop since if it fails to find those variables, they regress to their default variables here. That would necessitate them being in the script.

I don't really see any reason to define this stuff in the crontab. I think it opens another can of worms in terms of maintainence.

Kr1ss-XD commented:
Yeah, that makes sense !

I'll withdraw this PR then.
